ci: update ember-try config, add embroider test scenarios

yonmey opened the pull request:
## CI

### Update ember-try scenarios with Ember 3.24 LTS and Embroider tests (#121)

#### Remove `ember-` prefix from Github Actions config

#### Remove Ember 3.16 LTS from ember-try testing

#### Remove `ember-canary` scenario

#### Add Ember 3.24 LTS to ember-try testing

#### Add `@embroider/test-setup` to ember-try

This update adds the `embroider-safe` and `embroider-optimized` scenarios to ember-try. This gives the opportunity to check whether the addon is ready for Embroider or not.
[@embroider/test-setup](https://github.com/embroider-build/embroider/tree/master/packages/test-setup)

MrChocolatine commented:
@yonmey , testing against Ember.js beta and canary has been removed in #122 .

MrChocolatine commented on .github/workflows/ci.yml:
```diff
@@ -15,13 +15,12 @@
+#$   - scenario: lts-3.20
+#$   - scenario: lts-3.24
+#$   - scenario: release
+#$   - scenario: beta
```
Since you already removed the line `canary`.

```suggestion
```

MrChocolatine reviewed (commented):
We still have `*.orig` files do we want to keep them? These files also have `git-diff` syntax in them `<<<<<<< HEAD`, I am not sure we want to keep that.

Is the PR finished 🤔 ?

Also, why removing the prefix `ember-` from the test scenarios? It does not seem to bring any value here, also I thought it would be easier to have scenarios with a full name alongside Embroider scenarios named `embroider-*`.

yonmey commented:
> We still have `*.orig` files do we want to keep them? These files also have `git-diff` syntax in them `<<<<<<< HEAD`, I am not sure we want to keep that.
> Is the PR finished 🤔 ?

This has been a mistake, I have commit some files coming from the Idea `mergetool`, I didn't notice it. Fixed.

> Also, why removing the prefix `ember-` from the test scenarios? It does not seem to bring any value here, also I thought it would be easier to have scenarios with a full name alongside Embroider scenarios named `embroider-*`.

I removed the `ember-` prefix so that we have an homogeneous wording in all our projects, the Embroider config doesn't add this prefix.

MrChocolatine commented:
> I removed the `ember-` prefix so that we have an homogeneous wording in all our projects, the Embroider config doesn't add this prefix.

I see, I initially thought we wanted full names.
